src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParser.g4

```diff
@@ -889,8 +889,8 @@ javadocTag: AUTHOR_LITERAL (WS | NEWLINE)* ((WS | NEWLINE) description)?
 
       | RETURN_LITERAL (WS | NEWLINE)* ((WS | NEWLINE) description)?
```

Comment:
`a* (a b)?`
Can't this be rewritten as `a+ b?` ?
`SEE_LITERAL` might be similar too, maybe some of the others around it.

Reply:
Updated new expression to `(a+ b?)?`. Not sure what performance impact would be.

Reply:
i think it should be `| RETURN_LITERAL (WS | NEWLINE)+ description?`

Reply:
@rnveach 
I don't think that should happen. The performance reports didn't show anything like that. In fact it might be so that ANTLR produces _almost_ similar code for both cases. Though I can't say anthing for sure. WIll have too look deeper. 

@romani 
Yes I agree. We are doing this in #4942 . This issue split out from that one in the first place

Reply:
I see, new issue will require description. So really it should be like: `| RETURN_LITERAL (WS | NEWLINE)+ description` (no `?` on end like romani had).

Reply:
yes. Also, still there still shall be some places with construct like `a* (a b)?` for instance in `@deprecated` or `@customTag`. Replacing `(WS | NEWLINE)* ((WS | NEWLINE ) description)?`
with `((WS | NEWLINE )+ description)?` first of all will fail to take care of trailing `NEWLINES` so it will have to be something like `((WS | NEWLINE )+ description)? (WS | NEWLINE)*` anyway. 

Also, is some other construct possible which leads to greater efficiency ? perhaps not, at least the ANTLR report did not hint any performance degradation. We can discuss though if it can be made even better with some other construct.